### PR TITLE
[VEN-838] Added count of suppliers and borrowers to market

### PR DESCRIPTION
**Pull request by gleiser-oliveira**
## Jira ticket(s)

VEN-838

## Changes

- Added `supplierCount` to Market
- Added `borrowerCount` to Market
- Added unit and integration tests for the corresponding logic

### CODE REVIEW

File: subgraphs/isolated-pools/src/mappings/vToken.ts [hunk 1]
```diff
@@ -69,6 +80,15 @@ export function handleRedeem(event: Redeem): void {
   const vTokenAddress = event.address;
   const market = getMarket(vTokenAddress);
   createRedeemTransaction(event, market.underlyingDecimals);
+
+  // we read the current total amount of supplied tokens by this account in the market
+  const vTokenContract = VToken.bind(vTokenAddress);
+  const currentBalance = vTokenContract.balanceOf(event.params.redeemer);
+  if (currentBalance == zeroBigInt32) {
+    // and if they are the same, it means it's a new supplier
```
Comment (coreyar) [inline subgraphs/isolated-pools/src/mappings/vToken.ts:88]:
```suggestion
    // if the current balance is 0 then the user has withdrawn all their assets from this market
```

File: subgraphs/isolated-pools/src/mappings/vToken.ts [hunk 2]
```diff
@@ -51,6 +53,15 @@ export function handleMint(event: Mint): void {
   const vTokenAddress = event.address;
   const market = getMarket(vTokenAddress);
   createMintTransaction(event, market.underlyingDecimals);
+
+  // we read the current total amount of supplied tokens by this account in the market
+  const vTokenContract = VToken.bind(vTokenAddress);
+  const suppliedTotal = vTokenContract.balanceOf(event.params.minter);
```
Comment (coreyar) [inline subgraphs/isolated-pools/src/mappings/vToken.ts:59]: Should we see if we can update the event to save this call?

File: subgraphs/isolated-pools/src/mappings/vToken.ts [hunk 3]
```diff
@@ -69,6 +80,15 @@ export function handleRedeem(event: Redeem): void {
   const vTokenAddress = event.address;
   const market = getMarket(vTokenAddress);
   createRedeemTransaction(event, market.underlyingDecimals);
+
+  // we read the current total amount of supplied tokens by this account in the market
+  const vTokenContract = VToken.bind(vTokenAddress);
+  const currentBalance = vTokenContract.balanceOf(event.params.redeemer);
```
Comment (coreyar) [inline subgraphs/isolated-pools/src/mappings/vToken.ts:86]: Should we see if we can update the event to save this call?

Comment needs to be corrected

File: subgraphs/isolated-pools/src/mappings/vToken.ts (hunk 2 above)
Comment (gleiser-oliveira) [reply subgraphs/isolated-pools/src/mappings/vToken.ts:59]: Proposing this here https://github.com/VenusProtocol/isolated-pools/pull/144

File: subgraphs/isolated-pools/src/mappings/vToken.ts (hunk 3 above)
Comment (gleiser-oliveira) [reply subgraphs/isolated-pools/src/mappings/vToken.ts:86]: Proposing this here https://github.com/VenusProtocol/isolated-pools/pull/144

**Review by coreyar — APPROVED** [review]

**Pull request merged**